pymeasure/display/log.py

```diff
@@ -32,12 +32,16 @@
 log.addHandler(logging.NullHandler())
 
 
-class LogHandler(QtCore.QObject, Handler):
-    record = QtCore.QSignal(object)
+class LogHandler(Handler):
+    class Emitter(QtCore.QObject):
```

Comment:
Why the nested class? As you only end up using `record`, would it not be simpler to do `self.record = QtCore.QSignal(object)` in `LogHandler`s init, and use that?

Reply:
Qt signal needs to be class attribute, so this is why I was forced to create a dedicated class.
I preferred to put inside the LogHandler class to limit its visibility, since I do not see a general use for it.

Reply:
OK, I see. `record` already was a class attribute inside the `LogHandler` class, though?

Reply:
Indeed the original code was good for `Qt `signal management and for `logging` `emit `method.
The changes are required to avoid problems linked to conflict with `emit `method in `PySide2` signal management.
See also https://stackoverflow.com/questions/66664542/conflicting-names-between-logging-emit-function-and-qt-emit-signal

Reply:
I get the part about clobbering `emit`, and indeed that is resolved nicely. I just don't see how the `LogHandler` class attribute `record` is a part of that conflict, and we need to hide in in an `Emitter` class (and associated instance field `emitter`) that to me only seems to introduce additional API/interface surface for no reason I can discern. That's why I'm asking. :-)

Reply:
In other words, `self.emitter` does not do much and looks to me similar to a pass-through method.

Reply:
It's a bit weird, but an example can explain.
In `PySide2`, if an host class which contains a `Signal` class attribute and a method called `emit`, whenever you call the `emit` method of the `Signal` attribute, the host class method `emit` get called.
As a result, you can have either mismatch in the `emit` parameters or infinite recursion.
Have a look at the attached script, in particular, the class method `emit` method has a second parameter to avoid the signature mismatch.
The script goes into an infinite loop.
If the questions is why `PySide2` behaves like that then I do not know, but this is the best compromise I found to keep things running.

[test_pyside2.zip](https://github.com/pymeasure/pymeasure/files/6367586/test_pyside2.zip)

Reply:
Ah, that was the context I was missing, thanks for the explanation! Can you maybe add that as a commit to the `Emitter` class so that our future selves can get their bearings quickly? (see [Chesterton's fence](https://en.wikipedia.org/wiki/G._K._Chesterton#Chesterton's_fence))

Reply:
Added a comment to clarify purpose of class Emitter.